src/DateTimePicker.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import cx from 'classnames';
+import Flatpickr from 'flatpickr';
+
+/**
+ * @module DateTimePicker
+ */
+class DateTimePicker extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value || '',
+			isSupported: false
+		};
+		this.onChange = this.onChange.bind(this);
+	}
+
+	// test for browser support
+	componentWillMount() {
+		const input = document.createElement('input'),
+			invalidValue = 'notadate';
+
+		input.setAttribute('type', 'datetime-local');
+		input.setAttribute('value', invalidValue);
+
+		// adding false to view storybook
+		// in chrome since chrome supports
+		const isInvalidValueDenied = (false && input.value !== invalidValue);
+
+		// some browsers (as Android stock browsers) pretend they support
+		// certain input types, so set the value and see
+		this.setState({ isSupported: isInvalidValueDenied });
+	}
+
+	// init the js datetime component
+	// if there is no native support
+	componentDidMount() {
+		if (!this.state.isSupported) {
+			const options = {
+				...this.props.datepickerOptions,
+				onChange: this.onChange,
+				altInput: true,
+				altFormat: 'M d, Y h:i K' // TODO localize
+			};
+			this.flatpickr = new Flatpickr(this.node, options);
+		}
```

Comment:
a preferred pattern is to exit early rather than wrapping a whole function inside an `if` block:

```js
componentDidMount() {
  if (this.state.isSupported) {
    return;
  }
  // ...
}
```

it reduces nesting without hurting understandability